src/evm/WormholeVerifier.sol

```diff
@@ -1117,14 +1127,21 @@ contract WormholeVerifier is EIP712Encoding {
     // Verify the signature
     // We're not doing replay protection with the signature itself so we don't care about
     // verifying only canonical (low s) signatures.
+    bytes32 shardId = keccak256(abi.encode(pubKeyX, pubKeyY));
     bytes32 digest = getRegisterGuardianDigest(schnorrKeyIndex, nonce, shardId);
     address signatory = ecrecover(digest, v, r, s);
     require(signatory == expected, UpdateFailed(baseOffset | MASK_UPDATE_RESULT_SIGNATURE_MISMATCH));
 
     _useUnorderedNonce(schnorrKeyIndex, signerIndex, nonce, baseOffset);
 
+    (bytes32 oldPubKeyX, bytes32 oldPubKeyY) = _getSchnorrShardId(schnorrKeyIndex, signerIndex);
+
     // Store the shard ID
-    _setSchnorrShardId(schnorrKeyIndex, signerIndex, shardId);
+    _setSchnorrShardId(schnorrKeyIndex, signerIndex, pubKeyX, pubKeyY);
+    
+    bytes memory oldPublicKey = abi.encodePacked(oldPubKeyX, oldPubKeyY);
+    bytes memory newPublicKey = abi.encodePacked(pubKeyX, pubKeyY);
+    emit ShardIdUpdated(schnorrKeyIndex, signerIndex, oldPublicKey, newPublicKey);
```

Comment:
Hmm, I think I'd rather add four fixed size arguments, each representing a component to a pubkey, to the event instead of two dynamically sized keys. So you'd have:
- bytes 32 oldPubkeyX
- bytes 32 oldPubkeyY
- bytes 32 newPubkeyX
- bytes 32 newPubkeyY